datanode/dehistory/initialise.go

```diff
@@ -169,41 +173,46 @@ func FetchHistoryBlocks(ctx context.Context, logInfo func(s string, args ...inte
 	return blocksFetched, nil
 }
 
-func GetMostRecentHistorySegmentFromPeerAddresses(ctx context.Context, activePeerAddresses []string,
+type PeerResponse struct {
+	PeerAddr string
+	Response *v2.GetMostRecentDeHistorySegmentResponse
+}
+
+func GetMostRecentHistorySegmentFromPeersAddresses(ctx context.Context, peerAddresses []string,
+	swarmKey string,
 	grpcAPIPorts []int,
-) (*v2.HistorySegment, map[string]*v2.HistorySegment, error) {
+) (*PeerResponse, map[string]*v2.GetMostRecentDeHistorySegmentResponse, error) {
 	const maxPeersToContact = 10
 
-	if len(activePeerAddresses) > maxPeersToContact {
-		activePeerAddresses = activePeerAddresses[:maxPeersToContact]
+	if len(peerAddresses) > maxPeersToContact {
+		peerAddresses = peerAddresses[:maxPeersToContact]
 	}
 
 	ctxWithTimeOut, ctxCancelFn := context.WithTimeout(ctx, 30*time.Second)
 	defer ctxCancelFn()
-	peerToSegment := map[string]*v2.HistorySegment{}
+	peerToResponse := map[string]*v2.GetMostRecentDeHistorySegmentResponse{}
 	var errorMsgs []string
-	for _, peerAddress := range activePeerAddresses {
+	for _, peerAddress := range peerAddresses {
 		for _, grpcAPIPort := range grpcAPIPorts {
-			segment, err := GetMostRecentHistorySegmentFromPeer(ctxWithTimeOut, peerAddress, grpcAPIPort)
+			resp, err := GetMostRecentHistorySegmentFromPeer(ctxWithTimeOut, peerAddress, grpcAPIPort)
 			if err == nil {
-				if segment != nil {
-					peerToSegment[peerAddress] = segment
-				}
+				peerAddress = fmt.Sprintf("%s:%d", peerAddress, grpcAPIPort)
+				peerToResponse[peerAddress] = resp
 			} else {
 				errorMsgs = append(errorMsgs, err.Error())
 			}
 		}
 	}
 
-	if len(peerToSegment) == 0 && len(errorMsgs) != 0 {
-		return nil, nil, fmt.Errorf(strings.Join(errorMsgs, "\n"))
+	if len(peerToResponse) == 0 {
+		return &PeerResponse{}, nil, fmt.Errorf(strings.Join(errorMsgs, "\n"))
```

Comment:
why not return nil?

also and error with \n looks weird, maybe want to join with `", "` instead

Reply:
sorted